File: sdk/python/pe/resources/workflows.py
import httpx
from typing import Optional


class WorkflowsClient:
    def __init__(self, http: httpx.Client):
        self._http = http

    def templates(self) -> dict:
        return self._http.get("/workflows/templates").json()

    def template(self, name: str) -> dict:
        return self._http.get(f"/workflows/templates/{name}").json()

    def run(self, *, template_name: Optional[str] = None,
            template: Optional[dict] = None,
            vars: Optional[dict] = None) -> dict:
        body: dict = {}
        if template_name:
            body["templateName"] = template_name
        if template:
            body["template"] = template
        if vars:
            body["vars"] = vars
        return self._http.post("/workflows/run", json=body).json()

    def list(self, *, page: int = 1, page_size: int = 20) -> dict:
        return self._http.get("/workflows", params={"page": page, "pageSize": page_size}).json()

    def get(self, id: str) -> dict:
        return self._http.get(f"/workflows/{id}").json()

File: packages/sdk/src/resources/workflows.ts
import type { HttpClient } from '../http';
import type { WorkflowRun, WorkflowTemplate, WorkflowTemplateInfo, WorkflowRunsResponse } from '../types';

export interface RunWorkflowOptions {
  templateName?: string;
  template?: WorkflowTemplate;
  vars?: Record<string, unknown>;
}

export class WorkflowsClient {
  constructor(private readonly http: HttpClient) {}

  templates(): Promise<{ templates: WorkflowTemplateInfo[] }> {
    return this.http.get('/workflows/templates');
  }

  template(name: string): Promise<{ template: WorkflowTemplate }> {
    return this.http.get(`/workflows/templates/${name}`);
  }

  run(opts: RunWorkflowOptions): Promise<WorkflowRun> {
    return this.http.post<WorkflowRun>('/workflows/run', opts);
  }

  list(params: { page?: number; pageSize?: number } = {}): Promise<WorkflowRunsResponse> {
    const q = new URLSearchParams();
    if (params.page) q.set('page', String(params.page));
    if (params.pageSize) q.set('pageSize', String(params.pageSize));
    const qs = q.toString() ? `?${q}` : '';
    return this.http.get(`/workflows${qs}`);
  }

  get(id: string): Promise<WorkflowRun> {
    return this.http.get(`/workflows/${id}`);
  }
}
